File: results-05-12/typescript/typescript-00009-benchmark-store-website/qwen2-5-coder-32b/editor/final-src-code.ts

  // These environment variables are required, before any import.
  // Do not remove them, as they set environment variables for the Shinkai Tools.
  Deno.env.set('SHINKAI_NODE_LOCATION', "http://localhost:9950");
  Deno.env.set('BEARER', "debug");
  Deno.env.set('X_SHINKAI_TOOL_ID', "tool-id-debug");
  Deno.env.set('X_SHINKAI_APP_ID', "tool-app-debug");
  Deno.env.set('X_SHINKAI_LLM_PROVIDER', "o_qwen2_5_coder_32b");
  Deno.env.set('HOME', "results/typescript/typescript-00009-benchmark-store-website/qwen2-5-coder-32b/editor/home");
  Deno.env.set('MOUNT', "results/typescript/typescript-00009-benchmark-store-website/qwen2-5-coder-32b/editor/mount");
  Deno.env.set('ASSETS', "results/typescript/typescript-00009-benchmark-store-website/qwen2-5-coder-32b/editor/assets");
  
import { getHomePath } from './shinkai-local-support.ts';
import { shinkaiSqliteQueryExecutor } from './shinkai-local-tools.ts';

type CONFIG = {};
type INPUTS = { url: string };
type OUTPUT = { content: string };

export async function run(config: CONFIG, inputs: INPUTS): Promise<OUTPUT> {
    const homePath = getHomePath();
    const databaseName = 'default';
    
    // Create table if it doesn't exist
    await shinkaiSqliteQueryExecutor(databaseName, `
        CREATE TABLE IF NOT EXISTS website_content (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            url TEXT NOT NULL UNIQUE,
            content TEXT NOT NULL,
            last_updated DATETIME DEFAULT CURRENT_TIMESTAMP
        );
    `);

    // Fetch the website content
    const response = await fetch(inputs.url);
    if (!response.ok) {
        throw new Error(`Failed to fetch the website: ${inputs.url}`);
    }
    const content = await response.text();

    // Insert or update the website content in the database
    await shinkaiSqliteQueryExecutor(databaseName, `
        INSERT INTO website_content (url, content)
        VALUES (?, ?)
        ON CONFLICT(url) DO UPDATE SET content=excluded.content, last_updated=CURRENT_TIMESTAMP;
    `, [inputs.url, content]);

    // Retrieve the entire table
    const result = await shinkaiSqliteQueryExecutor(databaseName, 'SELECT * FROM website_content;');
    
    return { content: JSON.stringify(result.result) };
}

  
  // console.log('Running...')
  // console.log('Config: {}')
  // console.log('Inputs: {"url":"https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/test.html"}')
  
  try {
    const program_result = await run({}, {"url":"https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/test.html"});
    if (program_result) console.log(JSON.stringify(program_result, null, 2));
    else console.log(program_result);
  } catch (e) {
    console.log('::ERROR::', e);
  }

File: results-05-12/python/python-00002-benchmark-config/qwen2-5-coder-32b/editor/final-src-code.py

import os
import json
os.environ["SHINKAI_NODE_LOCATION"] = "http://localhost:9950"
os.environ["BEARER"] = "debug"
os.environ["X_SHINKAI_TOOL_ID"] = "tool-id-debug"
os.environ["X_SHINKAI_APP_ID"] = "tool-app-debug"
os.environ["X_SHINKAI_LLM_PROVIDER"] = "o_qwen2_5_coder_32b"
os.environ["HOME"] = "results/python/python-00002-benchmark-config/qwen2-5-coder-32b/editor/home"
os.environ["MOUNT"] = "results/python/python-00002-benchmark-config/qwen2-5-coder-32b/editor/mount"
os.environ["ASSETS"] = "results/python/python-00002-benchmark-config/qwen2-5-coder-32b/editor/assets"

from typing import Dict, Any, Optional, List
import requests

class CONFIG:
    api_key: str

class INPUTS:
    prompt: str

class OUTPUT:
    response: Optional[Dict[str, Any]]

async def run(config: CONFIG, inputs: INPUTS) -> OUTPUT:
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {config.api_key}"
    }
    
    data = {
        "model": "gpt-3.5-turbo",
        "messages": [{"role": "user", "content": inputs.prompt}]
    }
    
    response = requests.post("https://api.openai.com/v1/chat/completions", headers=headers, json=data)
    
    output = OUTPUT()
    if response.status_code == 200:
        output.response = response.json()
    else:
        output.response = None
    
    return output

if __name__ == "__main__":
    import asyncio
    
    config = CONFIG()
    inputs = INPUTS()
    inputs.prompt="2 + 2 = ?"
    
    # Run the async function
    result = asyncio.run(run(config, inputs))
    print(json.dumps(result.__dict__))
